charts/partials/templates/_proxy-init.tpl

```diff
@@ -58,19 +58,18 @@ securityContext:
     {{- include "partials.proxy-init.capabilities.drop" . | nindent 4 -}}
     {{- end }}
     {{- end }}
-  {{- if or .Values.proxyInit.closeWaitTimeoutSecs .Values.proxyInit.runAsRoot }}
-  {{- if .Values.proxyInit.closeWaitTimeoutSecs }}
-  privileged: true
-  {{- else }}
-  privileged: false
-  {{- end }}
+  {{- if .Values.proxyInit.runAsRoot }}
   runAsNonRoot: false
   runAsUser: 0
   {{- else }}
-  privileged: false
   runAsNonRoot: true
   runAsUser: {{ .Values.proxyInit.runAsUser | int | eq 0 | ternary 65534 .Values.proxyInit.runAsUser }}
   {{- end }}
+  {{- if or .Values.proxyInit.closeWaitTimeoutSecs .Values.proxyInit.privileged }}
+  privileged: true
+  {{- else }}
+  privileged: false
+  {{- end }}
```

Comment:
Another nit/layout one: might be better to leave this before the `runAs*` entries to keep the alphabetical ordering, which is how the k8s API also exposes things (and will also make the diff smaller).